src/root.zig

```diff
@@ -547,19 +580,14 @@ pub const Ascii = struct {
         penalty_case_mistmatch: i32 = -2,
     };
 
-    alg: Algorithm,
-    opts: Options,
-
     // public interface
 
     pub fn init(
         allocator: std.mem.Allocator,
-        max_haystack: usize,
-        max_needle: usize,
         opts: Options,
     ) !Ascii {
-        const alg = try Algorithm.init(allocator, max_haystack, max_needle);
-        return .{ .alg = alg, .opts = opts };
+        const alg = try Algorithm.init(allocator);
+        return Ascii{ .alg = alg, .opts = opts };
     }
```

Comment:
This should be reverted; it should still initialize with a maximum size and let the user call resize if they want to expand the size.

Reply:
Same, I will change it in #5